### PR TITLE
Custom import sprites with id

### PR DESCRIPTION
**Pull request by felipe2g**
This PR updates the logic for assigning IDs in the lists appearances.Outfit, appearances.Effect, appearances.Object, and appearances.Missile. The new logic checks if the ID already exists in the list before assigning it. If the ID is already present, a new one is generated based on the last element in the list.

Before:

```
appearance.Id = (uint)MainWindow.appearances.Outfit[^1].Id + 1;
```

After:

```
if (!MainWindow.appearances.Outfit.Any(a => a.Id == appearance.Id))
{
    appearance.Id = (uint)appearance.Id;
}
else
{
    appearance.Id = (uint)MainWindow.appearances.Outfit[^1].Id + 1;
}
```

Expected Impact:

Ensures duplicate IDs are avoided.
Maintains consistency across the four updated lists.

### CODE REVIEW

**Review by andersonfaaria — APPROVED** [review]

File: Assets Editor/DatEditor.xaml.cs [hunk 1]
```diff
@@ -436,122 +436,212 @@ private void LoadCurrentObjectAppearances()
...
-            A_FlagUpgradeClassificationAmount.Value = (CurrentObjectAppearance.Flags.Upgradeclassification != null && CurrentObjectAppearance.Flags.Upgradeclassification.HasUpgradeClassification) ? (int)CurrentObjectAppearance.Flags.Upgradeclassification.UpgradeClassification : 0;
-
-            NpcDataList.Clear();
-
-            if (CurrentObjectAppearance.Flags.Npcsaledata.Count > 0)
-            {
-                A_FlagNPC.IsChecked = true;
-                foreach (var npcdata in CurrentObjectAppearance.Flags.Npcsaledata)
-                    NpcDataList.Add(npcdata);
+            if (CurrentObjectAppearance == null || CurrentObjectAppearance.Flags == null)
```
Comment (Arch-Mina) [inline Assets Editor/DatEditor.xaml.cs:439]: Thanks for your contribution!
I see you've added a default value for non-existing flags. In the original code, we initialize the object to create these flags when they are absent.  is there a specific scenario you had in mind for this change?

Comment (felipe2g) [reply Assets Editor/DatEditor.xaml.cs:439]: Sometimes, when we convert an old spr/dat to assets, these variables might come as null, which was causing crashes (nullptr). So, in this PR, I thought of checking if this variable is null and assigning a default value. Some assets I tested were also missing these flags, which caused errors when trying to open them. But if you think it's unnecessary, I can remove it.

Comment (felipe2g) [reply Assets Editor/DatEditor.xaml.cs:439]: And i think it was a strategy to protect the sprites against people dont know how use assets editor in dev mode..

Comment (Arch-Mina) [reply Assets Editor/DatEditor.xaml.cs:439]: Yes, it's not necessary, would be great if you removed this part

Comment (felipe2g) [reply Assets Editor/DatEditor.xaml.cs]: Removed

**Pull request merged**